### DISCUSSION TITLE
'GetMessagesBatch' unit tests

### DISCUSSION THREAD

**Discussion by diogod3** [Farfetch/kafkaflow#264] [Q&A] [answered]
Hi everyone,

I'm writing a middleware that processes messages in batch, and I'm having some troubles writing unit tests for it, since I haven't figured out how to mock the 'IMessageContext' in a way where I can call the 'GetMessagesBatch' method. 

Has anyone faced a similar challenge? Is there any way to do this with mocks, or this isn't the way to go?

Thanks.

**Comment by filipeesch** [comment, marked as answer]
Hi @diogod3 , we are planning a refactor on BatchConsume to solve this problem in the future v3.
With the current implementation, you could create a class that receives a list of messages and the middleware calls your class, doing this way you can test your class directly.

**Reply by diogod3** [reply]
Yeah, that's what I ended up doing. Thanks.
